x/did/keeper/msg_server_did.go

```diff
@@ -80,23 +79,29 @@ func (m msgServer) DeactivateDID(goCtx context.Context, msg *types.MsgDeactivate
 
 }
 
-func VerifyDIDOwnership(signData *types.DIDDocument, seq uint64, doc *types.DIDDocument, verificationMethodID string, sig []byte) (uint64, error) {
-	verificationMethod, ok := doc.VerificationMethodFrom(doc.Authentications, verificationMethodID)
-	if !ok {
-		return 0, sdkerrors.Wrapf(types.ErrVerificationMethodIDNotFound, "VerificationMethodId: %s", verificationMethodID)
+func VerifyDIDOwnership(signData *types.DIDDocument, seq uint64, doc *types.DIDDocument, sig []byte) (uint64, error) {
+
+	docBz := doc.Document
+	document, err := ariesdid.ParseDocument(docBz)
+	if err != nil {
+		return 0, sdkerrors.Wrapf(types.ErrInvalidDIDDocument, "failed to parse did document")
 	}
 
+	// TODO: Currently, the public key is stored in the first verificationMethod and make signature using it.
+	//		 We will improve this later by using the document's proof field.
+	verificationMethod := document.VerificationMethod[0]
+
 	// TODO: Currently, only ES256K1 is supported to verify DID ownership.
 	//       It makes sense for now, since a DID is derived from a Secp256k1 public key.
 	//       But, need to support other key types (according to verificationMethod.Type).
 	if verificationMethod.Type != types.ES256K_2019 && verificationMethod.Type != types.ES256K_2018 {
 		return 0, sdkerrors.Wrapf(types.ErrVerificationMethodKeyTypeNotImplemented, "VerificationMethod: %v", verificationMethod.Type)
 	}
-	pubKeySecp256k1, err := secp256k1util.PubKeyFromBase58(verificationMethod.PublicKeyBase58)
-	if err != nil {
-		return 0, sdkerrors.Wrapf(types.ErrInvalidSecp256k1PublicKey, "PublicKey: %v", verificationMethod.PublicKeyBase58)
-	}
-	newSeq, ok := types.Verify(sig, signData, seq, pubKeySecp256k1)
+
+	var key secp256k1.PubKey = make([]byte, secp256k1.PubKeySize)
+	copy(key[:], document.VerificationMethod[0].Value)
```

Comment:
I think you can change simply like below :)
Or is there a reason you are copying a byte array?
```suggestion
	key := secp256k1.PubKey(document.VerificationMethod[0].Value)
```

Reply:
As I said https://github.com/medibloc/panacea-core/pull/626#discussion_r1098193126, this part will be change. Please review this part again later.

Reply:
OK.
If you change this, please ping to me

Reply:
In https://github.com/medibloc/panacea-core/pull/633, the keytype was changed to a btcec type.